File: frontend/app/commands.py
"""Chat command parsing and token-display helpers."""


def parse_token_command(
    text: str,
    set_token_command: str,
    clear_token_command: str,
) -> tuple[str | None, str | None]:
    """Parse token management commands from user input."""
    stripped = text.strip()
    if stripped == clear_token_command:
        return ("clear", None)
    if not stripped.startswith(f"{set_token_command} "):
        return (None, None)
    token = stripped[len(set_token_command) :].strip()
    return ("set", token)


def mask_token(token: str) -> str:
    """Mask a token value for safe user-visible confirmation text."""
    cleaned = token.strip()
    if len(cleaned) <= 10:
        return "*" * len(cleaned)
    return f"{cleaned[:6]}...{cleaned[-4:]}"

File: src/reactui/src/commands.ts
export function parseTokenCommand(
  text: string,
  setTokenCommand: string,
  clearTokenCommand: string,
): { kind: "set" | "clear" | null; token: string | null } {
  const stripped = text.trim();
  if (stripped === clearTokenCommand) {
    return { kind: "clear", token: null };
  }
  if (!stripped.startsWith(`${setTokenCommand} `)) {
    return { kind: null, token: null };
  }
  const token = stripped.slice(setTokenCommand.length).trim();
  return { kind: "set", token };
}

export function parsePersonaCommand(
  text: string,
  setPersonaCommand: string,
  clearPersonaCommand: string,
): { kind: "set" | "clear" | null; persona: string | null } {
  const stripped = text.trim();
  if (stripped === clearPersonaCommand) {
    return { kind: "clear", persona: null };
  }
  if (!stripped.startsWith(`${setPersonaCommand} `)) {
    return { kind: null, persona: null };
  }
  const persona = stripped.slice(setPersonaCommand.length).trim();
  return { kind: "set", persona };
}

export function maskToken(token: string): string {
  const cleaned = token.trim();
  if (cleaned.length <= 10) {
    return "*".repeat(cleaned.length);
  }
  return `${cleaned.slice(0, 6)}...${cleaned.slice(-4)}`;
}